cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -0,0 +1,1569 @@
+#![allow(non_snake_case)]
+
+use std::collections::HashMap;
+
+use crate::cdsl::encodings::{Encoding, EncodingBuilder};
+use crate::cdsl::instructions::{
+    BoundInstruction, InstSpec, Instruction, InstructionGroup, Predicate as InstructionPredicate,
+    PredicateNode as InstructionPredicateNode, PredicateRegistry,
+};
+use crate::cdsl::recipes::{EncodingRecipe, EncodingRecipeNumber, Recipes};
+use crate::cdsl::settings::{PredicateNumber as SettingsPredicateNumber, SettingGroup};
```

Comment:
`settings::PredicateNumber` gets renamed to `SettingsPredicateNumber` in several places. My own experience is that import renamings like this end up as one-more-thing that I have to remember when reading code and figuring out how all the parts relate to each other, so I have a preference for renaming at the source rather than at the import, however it's not a strong preference.